docs/cudf/source/developer_guide/testing.md

```diff
@@ -0,0 +1,143 @@
+# Testing cuDF
+
+## Tooling
+Tests in cuDF are written using [`pytest`](https://docs.pytest.org/en/latest/).
+Test coverage is measured using [`coverage.py`](https://coverage.readthedocs.io/en/6.4.1/),
+specifically the [`pytest-cov`](https://github.com/pytest-dev/pytest-cov) plugin.
+Code coverage reports are uploaded to [Codecov](https://app.codecov.io/gh/rapidsai/cudf).
+Each PR also indicates whether it increases or decreases in test coverage.
+
+## Test organization
+
+How tests are organized depends on which of the following two groups they fall into:
+
+1. Free functions operating on the above classes like `cudf.merge`.
+2. Methods of classes like `DataFrame` or `Series`.
+
+Tests of free functions should be grouped into files based on the API sections in our documentation.
+This places tests of similar functionality into the same testing module.
+Tests of class methods should be organized in the same way, except that this organization should be within a subdirectory corresponding to the class.
+For instance, tests of `DataFrame` indexing should be placed into `dataframe/test_indexing.py`.
+In cases where tests may be shared by multiple classes sharing a common parent (e.g. `DataFrame` and `Series` both require `IndexedFrame` tests),
+the tests may be placed in a directory corresponding to the parent class.
+
+## Test contents
+
+### Writing tests
+
+In general, functionality must be tested both for standard use cases and exceptional cases.
```

Comment:
```suggestion
In general, functionality must be tested for both standard use cases and exceptional cases.
```

Reply:
Prefer:

"In general, functionality must be tested for both standard and exceptional cases"

?

Reply:
I went with Lawrence's suggestion here since it's a superset of Bradley's proposed change and I like that wording.